sparksql-scalapb/src/main/scala/scalapb/spark/CatalystMapper.scala

```diff
@@ -0,0 +1,80 @@
+package scalapb.spark
+
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.expressions.objects.StaticInvoke
+import org.apache.spark.sql.types.{DataType, ObjectType, TimestampType}
+import scalapb.GeneratedMessageCompanion
+import scalapb.descriptors.{Descriptor, FieldDescriptor, PInt, PLong, PMessage, PValue}
+
+import scala.collection.immutable
+
+trait CatalystMapper {
```

Comment:
Instead of this trait,  I suggest we use `frameless.TypedEncoder[T]` (https://github.com/typelevel/frameless/blob/92ca70f9545f2e1aa2a62e93479e290aba818dcf/dataset/src/main/scala/frameless/TypedEncoder.scala#L18-L31) see change on `catalystMappers` in SchemaOptions.

Reply:
Maybe consider using `Injection`s which can be converted to TypedEncoders. Injections are defined here: https://github.com/typelevel/frameless/blob/92ca70f9545f2e1aa2a62e93479e290aba818dcf/core/src/main/scala/frameless/Injection.scala and are much easier to write and won't require the users to understand catalyst Expressions.

Reply:
Changed this trait to extends `TypedEncoder` because I still need the functionality to map `PMessage` to `Any`